### ISSUE TITLE
Bake Node, to combine Toonz Level information into a final, properly occluded node for processing

### ISSUE THREAD

**Issue by Bracket-H** [opentoonz/opentoonz#4366]
If you were to draw a circle on a toonz raster brush and filled it with an opaque color, then this filled circle would occlude things behind it.

If you were to add a few copies to the Xsheet and spread the copies out so they form an overlapping chain or something, then the stacking order would paint and render the final image with proper occlusion.

However, let's say you combined all those filled circles into an 'over' node so they get combined into one without doing anything but stack them.
And then after this 'over' node you added a palette filter, and only kept lines with the index 1.
Then even after combining them with the over node, or any other node (so far ...I haven't tried ALL blending modes)...

You will then see 5 full lineart rings, and not the linework with overlaps in mind.
Example pictures:
(The last image is the desired effect)
https://imgur.com/a/ddd5PrB

Why would this be a nice node, in my opinion?
Because then one could either apply effects to the lines in opentoonz, or export a lineart pass to be composited elsewhere, like natron or after effects or whatever.

And my ulterior motive for it is: Emulating old inked cels with scratchy brushes and the like. 
I can sort of do that already but it requires duplication of the macro effect I made for it to be applied to every column so the final image is occluded/overlapped properly.

But since scratchyness is a sort of 'global' fake effect (for example by texturizing it with some worn ink look all at once)
It would be nice to have a node that bakes the entire thing into one final, yet still-having-the-toonz-level-index-information at the end.

**Comment by Bracket-H** [comment]
I think this would be possible already, but the toonz level->texture node messes things up because it does some index filtering on its own.
A baked node would alleviate this too, I think.

**Comment by F-Burning** [comment]
The same thing happened to me as you, and after a loooong time experimenting, I realized that you can collapse the levels you want, and then use RGB key in that subxsheet to separate the line, and then apply effects, such as calligraphy. It works!

**Comment by Bracket-H** [comment]
That does seem to do the trick. Thanks!

But...maybe a bake node would still be nice, who knows what kind of edge case workarounds like these bring... hehe.

**Comment by Bracket-H** [comment]
Also, I realize I could just merge all the levels...but level merging seems to have some issues with ink indices and the like, too.
It's also more destructive than just a node that collects everything at some point in the chain...
 :pleading_face:

**Comment by F-Burning** [comment]
I see.

For me this way works fine, although It doesn't do much good though because you can't smooth the lines afterwards. but anything that improves workflow is welcome.

**Comment by Bracket-H** [comment]
Yeah, I hoped for a 'matte' workaround but I can't get it to work. 
I also looked into the FX code but that was too galaxy brain for me.

Smoothing and the like I have a better macro FX setup (what I've been showing off is just for illustration purposes)
Which involves blurs, sharpes, matte erosion and other things to both make the synthesized lines 'soft' but also saturated enough to be true blacks/darks.
A very specific range of pseudo antialiasing that clamps the strokes more or less 'just right'.

It's the plucking apart of the linework that puts the wrench in my system.
For the time being I'm simply duplicating it for the columns, but ...that's overkill and hard to maintain, even with linked effects and groups.

And again, a way to somehow bake a render into a toonz level would make compositing better.
Even if it were limited. As in, no special effects, only ink and paint results.

Multiple columns 
v
bake into a single cel either via a render or a node
v
use the baked results to composite fancy special effects like glows, shadows, etc
v
render as image sequence or video. 
(bonus/dream: rendering it into a preexisting 'industry' format that external compositors use.
Like exr, where opentoonz, just like blender and other software can add the passes as neatly separated things that can be composited/processed further)

Again, merging could work but just to extract linework? Mush everything together? Keep backups of the scene/levels pre merge? 
Juggle both sets? (Hide one, show the other)

So much hassle...especially for people who aren't a studio. I don't have an assistant who does this for me lol.
